tools/cbmc/proofs/Queue/QueueTakeMutexRecursive/QueueTakeMutexRecursive_harness.c

```diff
@@ -0,0 +1,84 @@
+/*
+ * FreeRTOS memory safety proofs with CBMC.
+ * Copyright (C) 2019 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+#include "FreeRTOS.h"
+#include "queue.h"
+#include "tasksStubs.h"
+#include "queue_datastructure.h"
+#include "cbmc.h"
+
+/* prvUnlockQueue is going to decrement this value to 0 in the loop.
+   We need a bound for the loop. Using 5 has a reasonable performance resulting
+   in 4 unwinding iterations of the loop. The loop is mostly modifying a
+   data structure in task.c that is not in the scope of the proof. */
+#ifndef PRV_UNLOCK_UNWINDING_BOUND
+	#define PRV_UNLOCK_UNWINDING_BOUND 5
+#endif
+
+/* Bound for the timeout cycles */
+#ifndef QueueSemaphoreTake_BOUND
```

Comment:
I don't really understand the logic of these defines. Some of them are written in the C file and others in `Makefile.json`. Is there a good reason for this? If not, then I can clean it up as part of the new build system rebase, no need to go through all of these right now IMHO.

Reply:
Deleting, and adding comments to Makefile.json.